addon/components/pix-input.hbs

```diff
@@ -16,21 +16,24 @@
   <div class="pix-input__container">
     <input
       id={{this.id}}
-      class={{if @errorMessage "pix-input__input--error"}}
+      class="pix-input__input--default {{this.validationStatusClassName}}"
```

Comment:
Ici `validationStatusClassName` va retourner une string vide ? J'ai un peu de mal à comprendre pourquoi on passe la méthode ici

Reply:
`validationStatusClassName` est une méthode que tu retrouves dans le fichier js du composant. 

Si le paramètre `validationStatus` (dans le js tu verras `this.args.validationStatus` est renseigné avec default, error ou success, `validationStatusClassName` va retourner une string comportant la class css de notre choix.

La liste est tout en haut du fichier js.
Si tu as besoin, on peut en parler de vive voix :)

Reply:
Lorsqu'on utilisera le composant dans une application, on va créer une méthode de validation (validateEmail par exemple pour vérifier que l'adresse entré par l'utilisateur est correcte)

Si la valeur est correcte, on fera en sorte que `validationStatus` soit à `success`.
Celle-ci va alors permettre à la méthode `validationStatusClassName`de retourner la bonne classe css attendu pour avoir le liseré vert sur le champ et le svg vert de check

Reply:
si par contre le `validationStatus` n'est pas renseigné, `validationStatusClassName` ne trouvera rien dans sa liste et dans ce cas il y a cette condition OU qui retournera rien. On gardera donc un comportement par défaut visuel du champ.

Reply:
oui c'est vrai que ça induit en erreur effectivement, je pourrais ne pas préciser default dans le css déjà inclut dans le composant en dur